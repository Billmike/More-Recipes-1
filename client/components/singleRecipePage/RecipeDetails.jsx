import React from 'react';
import PropTypes from 'prop-types';

import capitalizeFirstLetter from '../../utils/capitalizeFirstLetter';

const propTypes = {
  recipe: PropTypes.shape({
    category: PropTypes.string,
    description: PropTypes.string,
    preparationTime: PropTypes.number,
    ingredients: PropTypes.string,
    directions: PropTypes.string,
  }).isRequired,
};

/**
 * Function to format display of comma separated string values into lists
 *
 * @param {string} ingredients - The string value to be formated
 *
 * @returns {object} - React element represent the list item
 */
const formatIngredientsDisplay = ingredients => ingredients
  .split(',').map((ingredient, index) => (
    <li key={index.toString()}>{capitalizeFirstLetter(ingredient.trim())}</li>
  ));

/**
 * Function to format display of comma separated string values into lists
 *
 * @param {string} directions - The string value to be formated
 *
 * @returns {object} - React element represent the list item
 */
const formatDirectionsDisplay = directions => directions
  .split('.').map((direction, index) => (
    <li key={index.toString()}>{capitalizeFirstLetter(direction.trim())}</li>
  ));

/**
 * RecipeDetails component
 *
 * @param {object} props - The properties passed to the component
 *
 * @returns {object} React element
 */
const RecipeDetails = (props) => {
  const { recipe } = props;
  const {
    category, description, preparationTime, ingredients,
    directions
  } = recipe;

  return (
    <div className="row">
      <div className="col s12">
        <h6><b><em>Category:</em></b></h6>
        <p id="description" className="grey-text text-darken-3 lighten-3">
          {capitalizeFirstLetter(category)}
        </p>
      </div>
      <div className="col s12">
        <h6><b><em>Description:</em></b></h6>
        <p id="description" className="grey-text text-darken-3 lighten-3">
          {capitalizeFirstLetter(description)}
        </p>
      </div>
      <div className="col s12">
        <h6><b><em>Preparation Time (Minutes):</em></b></h6>
        <p id="prep-time" className="grey-text text-darken-3 lighten-3">
          {preparationTime}
        </p>
      </div>
      <div className="col s12">
        <h6><b><em>Ingredients:</em></b></h6>
        <ol id="ingredients" className="grey-text text-darken-3 lighten-3">
          {ingredients && formatIngredientsDisplay(ingredients)}
        </ol>
      </div>
      <div className="col s12">
        <h6><b><em>Directions:</em></b></h6>
        <ol id="directions" className="grey-text text-darken-3 lighten-3">
          {directions && formatDirectionsDisplay(directions)}
        </ol>
      </div>
    </div>
  );
};

RecipeDetails.propTypes = propTypes;

export default RecipeDetails;
