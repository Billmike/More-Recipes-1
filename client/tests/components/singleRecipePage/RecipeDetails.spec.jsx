import React from 'react';

import RecipeDetails from '../../../components/singleRecipePage/RecipeDetails';

describe('RecipeDetails component', () => {
  it('should render component', () => {
    const props = {
      recipe: {
        category: 'Lunch',
        description: 'Tasty dish',
        preparationTime: 55,
        ingredients: 'Rice, beans, tomatoes, salt',
        directions: 'Do this. Do this. Do that',
      }
    };
    const wrapper = shallow(<RecipeDetails {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
