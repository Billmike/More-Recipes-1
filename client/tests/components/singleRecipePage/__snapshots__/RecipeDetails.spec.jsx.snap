// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDetails component should render component 1`] = `
<div
  className="row"
>
  <div
    className="col s12"
  >
    <h6>
      <b>
        <em>
          Category:
        </em>
      </b>
    </h6>
    <p
      className="grey-text text-darken-3 lighten-3"
      id="description"
    >
      Lunch
    </p>
  </div>
  <div
    className="col s12"
  >
    <h6>
      <b>
        <em>
          Description:
        </em>
      </b>
    </h6>
    <p
      className="grey-text text-darken-3 lighten-3"
      id="description"
    >
      Tasty dish
    </p>
  </div>
  <div
    className="col s12"
  >
    <h6>
      <b>
        <em>
          Preparation Time (Minutes):
        </em>
      </b>
    </h6>
    <p
      className="grey-text text-darken-3 lighten-3"
      id="prep-time"
    >
      55
    </p>
  </div>
  <div
    className="col s12"
  >
    <h6>
      <b>
        <em>
          Ingredients:
        </em>
      </b>
    </h6>
    <ol
      className="grey-text text-darken-3 lighten-3"
      id="ingredients"
    >
      <li
        key="0"
      >
        Rice
      </li>
      <li
        key="1"
      >
        Beans
      </li>
      <li
        key="2"
      >
        Tomatoes
      </li>
      <li
        key="3"
      >
        Salt
      </li>
    </ol>
  </div>
  <div
    className="col s12"
  >
    <h6>
      <b>
        <em>
          Directions:
        </em>
      </b>
    </h6>
    <ol
      className="grey-text text-darken-3 lighten-3"
      id="directions"
    >
      <li
        key="0"
      >
        Do this
      </li>
      <li
        key="1"
      >
        Do this
      </li>
      <li
        key="2"
      >
        Do that
      </li>
    </ol>
  </div>
</div>
`;
